or-monte-carlo/src/com/waterloorocketry/openrocket_monte_carlo/SimulationData.java

```diff
@@ -7,165 +7,222 @@
-                .filter(e -> e.getType() == FlightEvent.Type.GROUND_HIT).findFirst()
-                .orElseThrow().getTime();
-        double apogeeTime = data.getTimeToApogee();
-
-        int apogeeIndex = Collections.binarySearch(time, apogeeTime);
-        int landingIndex = Collections.binarySearch(time, landingTime);
-        if (apogeeIndex < 0) {
-            throw new SimulationException("Time to apogee does not correspond to a valid index");
-        }
-        if (landingIndex < 0) {
-            throw new SimulationException("Time to landing does not correspond to a valid index");
-        }
+            double landingTime = branch.getEvents().stream()
+                    .filter(e -> e.getType() == FlightEvent.Type.GROUND_HIT).findFirst()
+                    .orElseThrow().getTime();
+            double apogeeTime = data.getTimeToApogee();
 
-        List<Double> stability = branch.get(FlightDataType.TYPE_STABILITY);
-
-        maxStability = branch.getMaximum(FlightDataType.TYPE_STABILITY);
-        double minStability = Double.NaN;
-        double initStability = Double.NaN;
-        for (int i = 0; i < time.size(); i++) {
-            Double s = stability.get(i);
-            // as per the previous implementation, stop considering stability 2s before apogee
-            // as well, stability will be NaN if the launch rod is not cleared or the forces are not
-            if (time.get(i) + 5.0 <= apogeeTime && !s.isNaN()) {
-                if (Double.isNaN(minStability) || s < minStability) {
-                    minStability = s;
-                }
+            int apogeeIndex = Collections.binarySearch(time, apogeeTime);
+            int landingIndex = Collections.binarySearch(time, landingTime);
+            if (apogeeIndex < 0) {
+                throw new SimulationException("Time to apogee does not correspond to a valid index");
             }
-            if (Double.isNaN(initStability) && !s.isNaN()) {
-                initStability = s;
+            if (landingIndex < 0) {
+                throw new SimulationException("Time to landing does not correspond to a valid index");
             }
-        }
-        this.minStability = minStability;
-        this.initStability = initStability;
 
-        this.apogeeStability = stability.get(apogeeIndex);
-        this.landingLatitude = lat.get(landingIndex);
-        this.landingLongitude = lng.get(landingIndex);
+            List<Double> stability = branch.get(FlightDataType.TYPE_STABILITY);
+
+            double minStability = Double.NaN;
+            double initStability = Double.NaN;
+            for (int i = 0; i < time.size(); i++) {
+                Double s = stability.get(i);
+                // as per the previous implementation, stop considering stability 2s before apogee
+                // as well, stability will be NaN if the launch rod is not cleared or the forces are not
+                if (time.get(i) + 10 <= apogeeTime && !s.isNaN()) {
```

Comment:
isn't this actually 10 seconds? (I don't recall how the time units work)

Reply:
You are right. Updated the comment, that was an old comment I forgot to change